IDE/AURIX/wolfBoot-tc3xx-wolfHSM/Cpu0_Main.c

```diff
@@ -0,0 +1,47 @@
+/* Cpu0_Main.c
+ *
+ * Copyright (C) 2014-2024 wolfSSL Inc.
+ *
+ * This file is part of wolfBoot.
+ *
+ * wolfBoot is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * wolfBoot is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with wolfBoot.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include "IfxCpu.h"
+#include "IfxScuWdt.h"
+#include "Ifx_Types.h"
+
+IFX_ALIGN(4) IfxCpu_syncEvent g_cpuSyncEvent = 0;
+
+extern void loader_main(void);
+
+void core0_main(void)
+{
+    IfxCpu_enableInterrupts();
+
+    /* !!WATCHDOG0 AND SAFETY WATCHDOG ARE DISABLED HERE!!
+     * Enable the watchdogs and service them periodically if it is required
+     */
+    IfxScuWdt_disableCpuWatchdog(IfxScuWdt_getCpuWatchdogPassword());
+    IfxScuWdt_disableSafetyWatchdog(IfxScuWdt_getSafetyWatchdogPassword());
+
+    /* Wait for CPU sync event */
+    IfxCpu_emitEvent(&g_cpuSyncEvent);
+    IfxCpu_waitEvent(&g_cpuSyncEvent, 1);
+
+    /* invoke wolfBoot */
+    loader_main();
+
+    while (1) {
```

Comment:
nope, everything is broken at this point and we don't want to do anything else. If we cause a boot loop it can sometimes be difficult to reattach an external debugger. IMO this is just a demo example so I don't think we need any elaborate error reporting or failover cases